### PR TITLE
Partial fix for 960

### PR DESCRIPTION
**Pull request by afourney**
## Why are these changes needed?

Partial fix for #960.  Make `agents` optional in many GroupChat help functions. Default to all agents (i.e., `self.agents`)

## Related issue number

#960 

## Checks

- [ ] I've included any doc changes needed for https://microsoft.github.io/autogen/. See https://microsoft.github.io/autogen/docs/Contribute#documentation to build and test documentation locally.
- [ ] I've added tests (if relevant) corresponding to the changes introduced in this PR.
- [ ] I've made sure all auto checks have passed.

### CODE REVIEW

File: autogen/agentchat/groupchat.py [hunk 1]
```diff
@@ -79,20 +81,26 @@ def next_agent(self, agent: Agent, agents: List[Agent]) -> Agent:
...
 
-    def select_speaker_prompt(self, agents: List[Agent]) -> str:
+    def select_speaker_prompt(self, agents: Optional[List[Agent]] = None) -> str:
         """Return the floating system prompt selecting the next speaker. This is always the *last* message in the context."""
+        if agents is None:
+            agents = self.agents
         return f"Read the above conversation. Then select the next role from {[agent.name for agent in agents]} to play. Only return the role."
 
-    def manual_select_speaker(self, agents: List[Agent]) -> Union[Agent, None]:
+    def manual_select_speaker(self, agents: Optional[List[Agent]]) -> Union[Agent, None]:
```
Comment (BeibinLi) [inline autogen/agentchat/groupchat.py:100]: Missing "= None" here?

Comment (afourney) [reply autogen/agentchat/groupchat.py:100]: Yup! Good catch :)

Comment (afourney) [reply autogen/agentchat/groupchat.py]: Fixed

**Review by BeibinLi — APPROVED** [review]
LGTM

**Comment by sonichi** [comment]
The added lines are not covered by test.

**Comment by qingyun-wu** [comment]
> The added lines are not covered by test.

Right. @afourney could you also add a test to cover the added code? Code coverage will be decreased otherwise as reported: "Comparison is base [(3a768c3)](https://app.codecov.io/gh/microsoft/autogen/commit/3a768c3189a95ffc86d989ec75fd0028e52d1961?el=desc&utm_medium=referral&utm_source=github&utm_content=comment&utm_campaign=pr+comments&utm_term=microsoft) 26.63% compared to head [(38034de)](https://app.codecov.io/gh/microsoft/autogen/pull/963?src=pr&el=desc&utm_medium=referral&utm_source=github&utm_content=comment&utm_campaign=pr+comments&utm_term=microsoft) 37.88%."

**Comment by afourney** [comment]
@qingyun-wu @sonichi I have added test coverage.

**Pull request merged**